src/ucp/api/ucp.h

```diff
@@ -429,13 +431,17 @@ enum ucp_am_cb_flags {
  * @ingroup UCP_WORKER
  * @brief Flags for sending a UCP Active Message.
  *
- * Flags dictate the behavior of ucp_am_send_nb
- * currently the only flag tells UCP to pass in
- * the sending endpoint to the call
- * back so a reply can be defined.
+ * Flags dictate the behavior of @ref ucp_am_send_nb and @ref ucp_am_send_nbx
+ * routines.
  */
 enum ucp_send_am_flags {
-    UCP_AM_SEND_REPLY = UCS_BIT(0)
+    UCP_AM_SEND_REPLY = UCS_BIT(0), /**< Force relevant reply endpoint to be
+                                         passed to the data callback on the
+                                         receiver. */
+    UCP_AM_SEND_EAGER = UCS_BIT(1), /**< Force UCP to use only eager protocol
+                                         for AM sends. */
+    UCP_AM_SEND_RNDV  = UCS_BIT(2)  /**< Force UCP to use only rendezvous protocol
```

Comment:
Just for clarity, rendezvous need not mean zero-copy, right? UCP would only and try and use a UCT zcopy operation but may decide later that it has to go through intermediate buffers. Is that correct? 

Also, I don't think eager and rndv have been defined in the header file.